packages/aws-cdk-lib/aws-events-targets/lib/lambda.ts

```diff
@@ -29,18 +39,32 @@ export class LambdaFunction implements events.IRuleTarget {
    * result from an EventBridge event.
    */
   public bind(rule: events.IRule, _id?: string): events.RuleTargetConfig {
+    const { deadLetterQueue, event, role } = this.props;
+
     // Allow handler to be called from rule
     addLambdaPermission(rule, this.handler);
 
-    if (this.props.deadLetterQueue) {
-      addToDeadLetterQueueResourcePolicy(rule, this.props.deadLetterQueue);
+    if (deadLetterQueue) {
+      addToDeadLetterQueueResourcePolicy(rule, deadLetterQueue);
+    }
+
+    if (role) {
+      role.addToPrincipalPolicy(this.invokeFunctionStatement());
     }
 
     return {
       ...bindBaseTargetConfig(this.props),
       arn: this.handler.functionArn,
-      input: this.props.event,
+      input: event,
+      role,
       targetResource: this.handler,
     };
   }
+
+  private invokeFunctionStatement() {
+    return new iam.PolicyStatement({
+      actions: ['lambda:InvokeFunction'],
+      resources: [this.handler.functionArn],
```

Comment:
I've followed the same pattern as the Event Bus target construct setting `actions` and `resources` only. I'm wondering whether we want to impose any `conditions` to lock this down further? For example, do we want to set a `aws:SourceArn` condition using the Rule ARN?